libs/core/src/lib/slider/slider.component.ts

```diff
@@ -453,13 +463,22 @@ export class SliderComponent implements OnInit, OnChanges, OnDestroy, ControlVal
                 handleIndex = SliderRangeHandles.Second;
             }
         }
+        if (this.isRtl()) {
+            if (KeyUtil.isKeyCode(event, LEFT_ARROW) || KeyUtil.isKeyCode(event, UP_ARROW)) {
+                newValue = prevValue + diff;
+            }
 
-        if (KeyUtil.isKeyCode(event, LEFT_ARROW) || KeyUtil.isKeyCode(event, DOWN_ARROW)) {
-            newValue = prevValue - diff;
-        }
+            if (KeyUtil.isKeyCode(event, RIGHT_ARROW) || KeyUtil.isKeyCode(event, DOWN_ARROW)) {
+                newValue = prevValue - diff;
+            }
+        } else {
+            if (KeyUtil.isKeyCode(event, LEFT_ARROW) || KeyUtil.isKeyCode(event, DOWN_ARROW)) {
+                newValue = prevValue - diff;
+            }
 
-        if (KeyUtil.isKeyCode(event, RIGHT_ARROW) || KeyUtil.isKeyCode(event, UP_ARROW)) {
-            newValue = prevValue + diff;
+            if (KeyUtil.isKeyCode(event, RIGHT_ARROW) || KeyUtil.isKeyCode(event, UP_ARROW)) {
+                newValue = prevValue + diff;
+            }
```

Comment:
Instead of an if-statement, it might be more efficient to create a multiplier value. For example:
```ts
            if (KeyUtil.isKeyCode(event, LEFT_ARROW) || KeyUtil.isKeyCode(event, UP_ARROW)) {
                newValue = prevValue + diff * rtlMultiplier;
            }
```
where
```ts
const rtlMultiplier = this.isRtl() ? -1 : 1;
```

Even better you could add `rtlMultiplier` as a class property and update it when the RTLService updates:

```ts
    constructor(
        private readonly _elementRef: ElementRef,
        private readonly _cdr: ChangeDetectorRef,
        private readonly _renderer: Renderer2,
        private readonly _platform: Platform,
        @Optional() private readonly _rtlService: RtlService,
        @Optional() private _contentDensityService: ContentDensityService
    ) {
        this._rtlService.subscribe(rtl => {
            this.rtlMultiplier = rtl ? -1 : 1;
        });
    }
```

Reply:
@KevinOkamoto I think this is not a feasible solution for the key press up and down arrow key press.

Reply:
ok I understand. Thanks.